### ISSUE TITLE
ENTERPRISE: "cannot load such file -- graphql-enterprise"

### ISSUE THREAD

**Issue by gaffney** [rmosolgo/graphql-ruby#4511]
**Describe the bug**

I cannot get `graphql-enterprise` to load.

**Versions**

`graphql`: 2.0.22
`graphql-pro`: 1.24.6
`graphql-enterprise`: 1.3.0

**GraphQL schema**

If this is needed, we can move this to email.

**Steps to reproduce**

The fastest way I reproduce is just running:

`bundle exec rake graphql:schema:dump`

This command works before I install `graphql-enterprise`, and fails after.

**Expected behavior**

The application would run without an error.

**Actual behavior**

<details>
<summary>Click to view exception backtrace</summary>

```
gaffney@Ryans-MacBook-Pro-2 backend % bundle exec rake graphql:schema:dump
rake aborted!
NameError: uninitialized constant GraphQL::ExecutionError
/Users/gaffney/.gem/ruby/3.0.4/gems/graphql-enterprise-1.3.0/lib/graphql/enterprise/active_operation_limiter.rb:29:in `<class:ActiveOperationLimiter>'
/Users/gaffney/.gem/ruby/3.0.4/gems/graphql-enterprise-1.3.0/lib/graphql/enterprise/active_operation_limiter.rb:23:in `<module:Enterprise>'
/Users/gaffney/.gem/ruby/3.0.4/gems/graphql-enterprise-1.3.0/lib/graphql/enterprise/active_operation_limiter.rb:14:in `<module:GraphQL>'
/Users/gaffney/.gem/ruby/3.0.4/gems/graphql-enterprise-1.3.0/lib/graphql/enterprise/active_operation_limiter.rb:13:in `<main>'
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/Users/gaffney/.gem/ruby/3.0.4/gems/graphql-enterprise-1.3.0/lib/graphql/enterprise.rb:3:in `<main>'
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/Users/gaffney/workspace/backend/config/application.rb:9:in `<top (required)>'
/Users/gaffney/workspace/backend/Rakefile:6:in `require_relative'
/Users/gaffney/workspace/backend/Rakefile:6:in `<top (required)>'
/Users/gaffney/.gem/ruby/3.0.4/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/gaffney/.rubies/ruby-3.0.4/bin/bundle:23:in `load'
/Users/gaffney/.rubies/ruby-3.0.4/bin/bundle:23:in `<main>'

Caused by:
LoadError: cannot load such file -- graphql-enterprise
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:15:in `require'
/Users/gaffney/.gem/ruby/3.0.4/gems/bootsnap-1.10.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:15:in `require'
/Users/gaffney/workspace/backend/config/application.rb:9:in `<top (required)>'
/Users/gaffney/workspace/backend/Rakefile:6:in `require_relative'
/Users/gaffney/workspace/backend/Rakefile:6:in `<top (required)>'
/Users/gaffney/.gem/ruby/3.0.4/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/gaffney/.rubies/ruby-3.0.4/bin/bundle:23:in `load'
/Users/gaffney/.rubies/ruby-3.0.4/bin/bundle:23:in `<main>'
(See full trace by running task with --trace)
```

</details>

**Additional context**

Relevant snippets from `Gemfile`:

```ruby
source "https://gems.graphql.pro/" do
  gem "graphql-enterprise"
  gem "graphql-pro"
end

gem "graphql", "2.0.22"
```

Relevant snippets from `Gemfile.lock`:

```ruby
GEM
  remote: https://gems.graphql.pro/
  specs:
    graphql-enterprise (1.3.0)
      graphql (>= 1.13.0)
      graphql-pro (>= 1.24.0)
    graphql-pro (1.24.6)
      graphql (>= 1.7.5)

GEM
  remote: https://rubygems.org/
  specs:
    graphql (2.0.22)

DEPENDENCIES
  graphql (= 2.0.22)
  graphql-enterprise!
  graphql-pro!
```

**Comment by gaffney** [comment]
Alright, came back to this and found the workaround is to put the:

```ruby
source "https://gems.graphql.pro/" do
  gem "graphql-enterprise"
  gem "graphql-pro"
end
```

**below** `source "https://rubygems.org"` → `gem "graphql", "2.0.22"`.

However, I still suspect this is a bug, because we have the exact same situation with Sidekiq and do not run into any issues.

That is:
```ruby
# frozen_string_literal: true

source "https://enterprise.contribsys.com/" do
  gem "sidekiq-ent"
  gem "sidekiq-pro"
end
```

is at the top of our Gemfile, and `gem "sidekiq", "~> 7"` is below under `source "https://rubygems.org`.

**Comment by rmosolgo** [comment]
Sorry for the trouble, and thanks for the detailed report! I think I was missing: 

```ruby 
require "graphql"
```

At the top of `graphql/enterprise.rb`. So, I just released v1.3.1 with that fix. Could you give it a try and see if it still works after you re-order your gemfile? 

(Judging by the `Gemfile.lock`,  it seems like bundler _was_ using the right source for each gem, but graphql-enterprise was (wrongly) requiring `graphql` to come first in the Gemfile so that bundler would auto-load it first.)

**Comment by rmosolgo** [comment]
Please let me know if you encounter this issue on v1.3.1+!
